coreos-installer

```diff
@@ -317,6 +317,25 @@ write_ignition_file() {
     cp /tmp/ignition.ign /mnt/boot_partition/ignition/config.ign
 }
 
+############################################################
+# Helper to persist networking options to initramfs first boot
+############################################################
+write_networking_opts() {
+    if [ ! -f /tmp/ip_opt ]; then
+        return
+    fi
+
+    dialog --title 'CoreOS Installer' --infobox "Embedding provided networking options" 5 70
+    # check for the boot partition
+    mkdir -p /mnt/boot_partition
+    # TODO check to make sure the disk with label 'boot'
+    #      is part of ${DEST_DEV}
+    mount /dev/disk/by-label/boot /mnt/boot_partition
+    trap 'umount /mnt/boot_partition' RETURN
```

Comment:
Now that we're mounting/unmounting in two places in this file I'll ask a dumb question: does this unmount on return of this function?

Reply:
I believe so, https://ss64.com/bash/trap.html directly calls out shell functions for `RETURN` and testing inside of a shell script locally seems to work. I'll make another note in this comment chain about it when I test this properly.